apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.js

```diff
@@ -29,25 +26,30 @@ async function ensureUpToDate(locals) {
 router.post(
   '/',
   asyncHandler(async function (req, res, next) {
-    if (!res.locals.authz_result.authorized_edit) {
+    if (
+      !res.locals.authz_result.authorized_edit &&
+      !res.locals.authz_data.has_course_instance_permission_edit
+    ) {
       throw error.make(403, 'Not authorized', res.locals);
     }
+    if (
+      !res.locals.authz_result.authorized_edit &&
+      ['attach_file', 'attach_text', 'delete_file', 'timeLimitFinish', 'leave_group'].includes(
+        req.body.__action,
+      )
+    ) {
+      throw error.make(403, 'Action is only permitted to students, not staff', res.locals);
+    }
```

Comment:
This is new; I'm curious why this ended up in this PR? I don't see the relation to everything else happening here.

Reply:
There was a generic check on the lines before that blocked any POST when `authorized_edit` was false. However, I want to allow instructors to update roles, so I separated into two checks (the one before this and this one): the first causes a 403 if the user is neither an authorized student or a staff with edit permission; the second blocks retains the previous behaviour that blocks a staff member from posting for these particular kinds of actions.